services/src/main/java/org/keycloak/services/resources/account/AccountFormService.java

```diff
@@ -982,9 +982,8 @@ public static boolean isPasswordSet(KeycloakSession session, RealmModel realm, U
                 return new String[]{referrerName, referrerUri};
             }
         } else if (referrerUri != null) {
-            referrerClient = realm.getClientByClientId(referrer);
             if (client != null) {
-                referrerUri = RedirectUtils.verifyRedirectUri(session.getContext().getUri(), referrerUri, realm, referrerClient);
```

Comment:
At this time `referrerClient` is always null, and it should be changed to `client`.